src/main/scala/org/opennetworkinsight/netflow/FlowPreLDA.scala

```diff
@@ -23,186 +21,126 @@ object FlowPreLDA {
         + rowsToDuplicate.length + " many connections flagged nonthreatening.")
-      logger.info("Duplication factor: " + duplicationFactor)
-      rowsToDuplicate.map(convert_feedback_row_to_flow_row(_)).flatMap(List.fill(duplicationFactor)(_))
+
+      val feedback: RDD[String] = sc.parallelize(rowsToDuplicate)
+
+      feedback.map(_.split(","))
+        .filter(row => row.length == 22 && row(scoreIndex).trim.toInt == 3)
+        .map(row => row :+
+          row(timeStartIndex).split(" ")(1).split(":")(0) :+
+          row(timeStartIndex).split(" ")(1).split(":")(1) :+
+          row(timeStartIndex).split(" ")(1).split(":")(2))
+        .map(row => Feedback(row(hourIndex).trim.toInt,
+          row(minuteIndex).trim.toInt,
+          row(secondIndex).trim.toInt,
+          row(sourceIpIndex),
+          row(destinationIpIndex),
+          row(sourcePortIndex),
+          row(destinationPortIndex),
+          row(ipktIndex).trim.toLong,
+          row(ibytIndex).trim.toLong))
+        .flatMap(row => List.fill(duplicationFactor)(row))
+        .toDF()
     } else {
-      Array[String]()
+      null
     }
 
-    val totalData: RDD[String] = rawdata.union(sc.parallelize(scoredData))
-
-    val data_with_time = totalData.map(_.trim.split(',')).map(FlowWordCreation.addTime)
+    logger.info("Trying to read file:  " + inputPath)
+    val rawdata: DataFrame = {
+      sqlContext.read.parquet(inputPath)
+        .filter(Schema.Hour + " BETWEEN 0 AND 23 AND  " +
+          Schema.Minute + " BETWEEN 0 AND 59 AND  " +
+          Schema.Second + " BETWEEN 0 AND 59")
+        .select(Schema.Hour,
+          Schema.Minute,
+          Schema.Second,
+          Schema.SourceIP,
+          Schema.DestinationIP,
+          Schema.SourcePort,
+          Schema.DestinationPort,
+          Schema.ipkt,
+          Schema.ibyt)
+    }
 
-    logger.info("calculating time cuts ...")
-    time_cuts = Quantiles.computeDeciles(data_with_time.map(row => row(indexOf.NUMTIME).toDouble))
-    logger.info(time_cuts.mkString(","))
-    logger.info("calculating byte cuts ...")
-    ibyt_cuts = Quantiles.computeDeciles(data_with_time.map(row => row(indexOf.IBYT).toDouble))
-    logger.info(ibyt_cuts.mkString(","))
-    logger.info("calculating pkt cuts")
-    ipkt_cuts = Quantiles.computeQuintiles(data_with_time.map(row => row(indexOf.IPKT).toDouble))
-    logger.info(ipkt_cuts.mkString(","))
+    val totalDataDF: DataFrame = {
+      if (!scoredFileExists) {
+        rawdata
+      } else {
+        rawdata.unionAll(falsePositives)
+      }
+    }
 
-    val binned_data = data_with_time.map(row => FlowWordCreation.binIbytIpktTime(row, ibyt_cuts, ipkt_cuts, time_cuts))
+    val dataWithWordDF = FlowWordCreation.flowWordCreation(totalDataDF, sc, logger, sqlContext)
 
-    val data_with_words = binned_data.map(row => FlowWordCreation.adjustPort(row))
+    val src_word_counts = dataWithWordDF.select(Schema.SourceIP, Schema.SourceWord)
+      .map({
```

Comment:
the formatting of this map is confusing and it actually all fits easily into 120 character line